webapp/lib/controller.dart

```diff
@@ -108,71 +121,124 @@ void command(UIAction action, Data actionData) {
+  view.contentView.needsReplyAndEscalateLatestValue.updateChart('${latestData.needsReplyAndEscalateCount}');
+  view.contentView.needsReplyMoreThan24hLatestValue.updateChart('${latestData.needsReplyMoreThan24h}');
+  view.contentView.needsReplyAndEscalateMoreThan24hLatestValue.updateChart('${latestData.needsReplyAndEscalateMoreThan24hCount}');
 
-      view.contentView.needsReplyAgeHistogram.updateChart(latestData.needsReplyMessagesByDate);
-      break;
+  view.contentView.needsReplyAgeHistogram.updateChart(latestData.needsReplyMessagesByDate);
+}
 
-    case UIAction.systemEventsDataUpdated:
-      if (actionData != null && (actionData as ProjectSelectionData).isProjectSelection) {
-        view.contentView.populateUrlFilters();
-      } else {
-        view.contentView.changeViewOnUrlChange();
-      }
-      var rapidProEventData = systemEventsDataList.where((eventData) =>
-          eventData.systemName == 'rapidpro_adapter' &&
-          eventData.project == view.contentView.projectSelectorView.selectedProject);
-      var pubsubEventData = systemEventsDataList.where((eventData) =>
-          eventData.systemName == 'pubsub_handler' &&
-          eventData.project == view.contentView.projectSelectorView.selectedProject);
-      
-      Map<DateTime, int> data = new Map.fromIterable(rapidProEventData,
-          key: (item) => (item as model.SystemEventsData).timestamp,
-          value: (item) => 1);
-      view.contentView.rapidProSystemEventTimeseries.updateChart([data]);
-
-      data = new Map.fromIterable(pubsubEventData,
-          key: (item) => (item as model.SystemEventsData).timestamp,
-          value: (item) => 1);
-      view.contentView.pubsubSystemEventTimeseries.updateChart([data]);
+void updateSystemEventsCharts(List<model.SystemEventsData> filteredSystemEventsDataList) {
+  var rapidProEventData = filteredSystemEventsDataList.where((eventData) =>
+      eventData.systemName == 'rapidpro_adapter' &&
+      eventData.project == view.contentView.projectSelectorView.selectedProject);
+  var pubsubEventData = filteredSystemEventsDataList.where((eventData) =>
+      eventData.systemName == 'pubsub_handler' &&
+      eventData.project == view.contentView.projectSelectorView.selectedProject);
+  
+  Map<DateTime, int> data = new Map.fromIterable(rapidProEventData,
+      key: (item) => (item as model.SystemEventsData).timestamp,
+      value: (item) => 1);
+  view.contentView.rapidProSystemEventTimeseries.updateChart([data]);
+
+  data = new Map.fromIterable(pubsubEventData,
+      key: (item) => (item as model.SystemEventsData).timestamp,
+      value: (item) => 1);
+  view.contentView.pubsubSystemEventTimeseries.updateChart([data]);
+}
+
+DateTime getFilteredDate(ChartFilterdata filterData) {
+  DateTime now = new DateTime.now();
+  DateTime filterDate;
+
+  switch (filterData.periodFilter) {
+    case ChartPeriodFilters.alltime:
+      filterData = null;
```

Comment:
is there a typo here? should this be `filterDate` rather than `filterData`?

Reply:
Fixed this. Changes in new [PR](https://github.com/AfricasVoices/nook_watcher/pull/17)